app/public/script.js

```diff
@@ -1,12 +1,51 @@
+const e = require("express");
+
 const webSocket = new WebSocket(`ws://localhost:${WEBSOCKET_PORT}/`);
 webSocket.onmessage = (event) => {
   console.log(event.data);
 };
 webSocket.addEventListener("open", () => {
   console.log("We are connected");
+  sendMessage("Hi from client");
 });
-const sendMessage = () => {
+
+const sendMessage = (message) => {
   // Sends message to websocket server
-  webSocket.send("Hi from client");
+  // "Hi from client"
+  webSocket.send(message);
+};
+
+document
+  .getElementById("loginForm")
+  .addEventListener("submit", function (event) {
+    event.preventDefault();
+
+    let username = document.getElementById("username").value;
+    let password = document.getElementById("password").value;
+    loginUser(username, password);
+  });
+
+function loginUser(username, password) {
+  fetch("/login", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      username: username,
+      password: password,
+    }),
+  })
```

Comment:
The promise below has a catch, but this initial POST fetch needs a catch too. Every promise needs a catch.

For example, I'm now getting a 404 response error from this fetch because there's no "/login" route defined on the server. There will eventually be one, but for now we need to output an error message to the client.